SonarQube.Common/ProcessRunnerArguments.cs

```diff
@@ -128,18 +128,59 @@ public static bool ContainsSensitiveData(string text)
         }
 
         /// <summary>
-        /// According to https://msdn.microsoft.com/en-us/library/system.diagnostics.processstartinfo.arguments(v=vs.110).aspx
-        /// quotes have to be doubled. Also, to avoid problems when the argument has spaces or chars such as & which are problematic in .bat files, 
-        /// enclose the entire argument in quotes
+        /// The CreateProcess Win32 API call only takes 1 string for all arguments.
+        /// Ultimately, it is the responsibility of each program to decide how to split this string into multiple arguments.
+        ///
+        /// See:
+        /// https://blogs.msdn.microsoft.com/oldnewthing/20100917-00/?p=12833/
+        /// https://blogs.msdn.microsoft.com/twistylittlepassagesallalike/2011/04/23/everyone-quotes-command-line-arguments-the-wrong-way/
+        /// http://www.daviddeley.com/autohotkey/parameters/parameters.htm
```

Comment:
I think the probably of that is really low @bgavrilMS 
We've spent a lot of time to understand exactly how arguments are passed :)

The previous algo failed only when one of the arguments had a trailing backslash, in which case it would merge all following arguments into 1